internal/configuration/root/root.go

```diff
@@ -104,6 +105,29 @@ func initConfigOption() {
 	}
 }
 
+func initDetailedExitCodeOption() {
+	cobraParamName := "detailed-exitcode"
+	cobraValue := new(customtypes.Bool)
+	defaultValue := customtypes.Bool(false)
+
+	options.RootDetailedExitCodeOption = options.Option{
+		CobraParamName:  cobraParamName,
+		CobraParamValue: cobraValue,
+		DefaultValue:    &defaultValue,
+		EnvVar:          "PINGCLI_DETAILED_EXITCODE",
+		Flag: &pflag.Flag{
+			Name:        cobraParamName,
+			Shorthand:   "D",
+			Usage:       "Enable detailed exit code output. (default false)",
```

Comment:
Maybe add breakdown of possible exit codes here

1. 0 - pingcli command succeeded with no errors or warnings
2. 1 - pingcli command failed with errors
3. 2 - pingcli command succeeded with warnings